jdaviz/configs/imviz/wcs_utils.py

```diff
@@ -317,15 +303,17 @@ def rotated_gwcs(
 
     # "rescale" the pixel scales. Scaling constant was tuned so that the
     # synthetic image is about the same size on the sky as the input image
+    # for some arbitrary test data. If need re-tuning, ask Brett Morris.
     rescale_pixel_scale = np.array(refdata_shape) / 1000
 
     shift_by_crpix = (
         models.Shift(-refdata_shape[0] / 2 * u.pixel) &
         models.Shift(-refdata_shape[1] / 2 * u.pixel)
     )
 
-    # multiplying by +/-1 can flip north/south or east/west
-    flip_axes = models.Multiply(cdelt_signs[0]) & models.Multiply(cdelt_signs[1])
+    # Multiplying by +/-1 can flip north/south or east/west.
+    e_sign = -1  # We want E-left by default.

```

Comment:
This might be subtle enough that you might miss it scrolling through. Does this look correct? I was playing with different ways to make N-up E-left and this was what worked for me.

Reply:
In my concept notebook, without this sign flip, it rotates not the way I thought it would when the angle is `0 * u.deg`. But I don't understand the subtleties of all the different WCS conventions out there. The WCS I use in the concept notebook is the same as the ones here:

https://github.com/spacetelescope/jdaviz/blob/135ae31c0acad06abe9af60d5bbbfa8eb1c29010/jdaviz/configs/imviz/tests/utils.py#L98

Reply:
I guess I'm wondering how you know what you expect. The data I'm loading have a ground truth (sky truth?) that you can check, i.e. [here](https://aladin.cds.unistra.fr/AladinLite/?target=05%2040%2059.000-02%2027%2030.00&fov=0.82&survey=CDS%2FP%2FDSS2%2Fcolor). If you run my notebook with `rot=0*u.deg`, you see the same orientation as Aladin.

Reply:
> how you know what you expect

The problem is I don't. I naively thought simply passing in `0 * u.deg` would give me N-up E-left no matter what. Isn't that what users would expect too? We don't know what WCS they will have, but they expect zero angle means certain sky orientation.

Reply:
I could undo this sign change for now, if you want. This is more math than infrastructure.

Reply:
Hmm is that what we want though? I thought having presets (eventually) would mean 0 means N-up, and so on? How would I know what angle means N-up without also doing a bunch of WCS math depending on what WCS is the input?

Reply:
This math is already implemented in canvas rotation here:

https://github.com/spacetelescope/jdaviz/blob/acbaf3089d768ddec297c4ff7c7f0ca63edec2ac/jdaviz/configs/imviz/plugins/rotate_canvas/rotate_canvas.py#L74-L106

so I intentionally didn't rewrite those methods. The new plugin that we will write will call the methods in my PR, and use the canvas rotation methods (or their generalizations) to go to presets like N-up.

Reply:
Can you please clarify a little more here? For instance, once you have `degn`, do you pass it in as-is to `_get_rotated_nddata_from_label` or you pass in `-degn`? I just want to understand how all these parts are going to play together.

Reply:
I haven't worked with those methods yet so I don't already know. It looks like [get_compass_info](https://github.com/spacetelescope/jdaviz/blob/acbaf3089d768ddec297c4ff7c7f0ca63edec2ac/jdaviz/configs/imviz/wcs_utils.py#L118) will return the angle from the +y direction of your reference data image towards North as `degn`, though it's unclear from the docstring if that's the CW or CCW angle. So your suggestion above is probably correct or needs a `*(-1)`.

Reply:
I thought I did that yesterday but it also flipped in the E-W direction for some reason. Hmm. Anyways, I guess I can revert this and chuck a "FIXME" note about this.

Reply:
Though if you already know what sky orientation you want, might be easier to hardcode the WCS than to do extra math. I think a WCS with N-up E-left is pretty trivial to make by hand, thought I did get stuck on the pixel scale part which you solved, so I dunno... have to think about it maybe as a follow-up work.

Reply:
To follow up, I've confirmed that this particular sign flip in your PR breaks the [intended behavior](https://github.com/bmorris3/jdaviz/pull/4#discussion_r1196892119). Please revert.